### PR TITLE
first draft of concept event protocol

### PR DESCRIPTION
**Pull request by clauspeterhuebner**
prüf mal bitte das Konzept BusinessEventProtocol.md, ob das so verständlich ist und für die Umsetzung ausreicht.
Dann gibt es dazu zwei Implementierungsticket #1794 und #1795.

Ich nehm Ulf mal noch mit in den Review...

### CODE REVIEW

**Review by Mogge — DISMISSED** [review]
Very good first draft. Very helpful to create the database table

File: docu/Concepts/TechnicalRequirements/BusinessEventProtocol.md [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+| id           | int       | technical unique key (from db sequence)                                                          |
+| type         | enum      | type of event                                                                                    |
+| createdAt    | timestamp | timestamp the event occurs (not the time of writing)                                             |
+| userID       | string    | the user ID, who invokes the event                                                               |
+| XuserID      | string    | the cross user ID, who is involved in the process like a tx-sender, contrib-receiver, ...        |
+| XcommunityID | string    | the cross community ID, which is involved in the process like a tx-sender, contrib-receiver, ... |
+| txID         | int       | the technical key of the transaction, which triggers the event                                   |
+| contribID    | int       | the technical key of the contribution, which triggers the event                                  |
+| amount       | digital   | the amount of gradido transferred by transaction, creation or redeem                             |
```
Comment (ulfgebhardt) [inline docu/Concepts/TechnicalRequirements/BusinessEventProtocol.md:49]: we might also want a generic `reference` or `refID` field, which holds something like the `transactionID` or similar.

Comment (clauspeterhuebner) [reply docu/Concepts/TechnicalRequirements/BusinessEventProtocol.md:49]: the transactionID exists in form of the attribut txID.
for sure one or more general usable attribute could be added.
For the predefined list of event types all relevant values can be stored in the event protocol table

File: docu/Concepts/TechnicalRequirements/BusinessEventProtocol.md [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+| id            | int       | technical unique key (from db sequence)                                                          |
+| type          | enum      | type of event                                                                                    |
+| createdAt     | timestamp | timestamp the event occurs (not the time of writing)                                             |
+| userID        | string    | the user ID, who invokes the event                                                               |
+| XuserID       | string    | the cross user ID, who is involved in the process like a tx-sender, contrib-receiver, ...        |
+| XcommunityID  | string    | the cross community ID, which is involved in the process like a tx-sender, contrib-receiver, ... |
+| transactionID | int       | the technical key of the transaction, which triggers the event                                   |
+| contribID     | int       | the technical key of the contribution, which triggers the event                                  |
+| amount        | digital   | the amount of gradido transferred by transaction, creation or redeem                             |
```
Comment (Mogge) [inline docu/Concepts/TechnicalRequirements/BusinessEventProtocol.md:47]: Do we want to track success state of the events? Do we want to save thrown errors of an event?

**Review by Mogge — APPROVED** [review]
I suggest to merge. When we start working on the events, we will take these documents as base for our discussion and modify them if necessary

**Pull request merged**